src/WorkflowCore/Services/WorkflowExecutor.cs

```diff
@@ -211,25 +211,24 @@ private void DetermineNextExecutionTime(WorkflowInstance workflow)
             if (workflow.Status == WorkflowStatus.Complete)
                 return;
 
-            foreach (var pointer in workflow.ExecutionPointers.Where(x => x.Active && (x.Children ?? new List<string>()).Count == 0))
+            foreach (var pointer in workflow.ExecutionPointers.Where(x => x.Active))
             {
-                if (!pointer.SleepUntil.HasValue)
+                if ((pointer.Children ?? new List<string>()).Count == 0)
                 {
-                    workflow.NextExecution = 0;
-                    return;
-                }
-
-                var pointerSleep = pointer.SleepUntil.Value.ToUniversalTime().Ticks;
-                workflow.NextExecution = Math.Min(pointerSleep, workflow.NextExecution ?? pointerSleep);
-            }
+                    if (!pointer.SleepUntil.HasValue)
+                    {
+                        workflow.NextExecution = 0;
+                        return;
+                    }
 
-            if (workflow.NextExecution == null)
```

Comment:
I think the only required change is to remove this `if` statement and always execute the inner code.
We can remove all the other changes... seems like an attempt for optimization that is causing a problem.

Reply:
@andre-pt this is a valid bug that needs to be fixed.  Do you want to do it with this PR or should I open a separate one?

Reply:
https://github.com/danielgerlag/workflow-core/pull/700